pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param href: A string. The path to the repository to which a distributor
+        shall be added.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return handle_response(requests.post(
+        server_config.base_url + href + 'distributors/',
+        json={
+            'auto_publish': False,
+            'distributor_id': uuid4(),
+            'distributor_type_id': 'puppet_distributor',
+            'distributor_config': {
+                'serve_http': True,
+                'serve_https': True,
+                'relative_url': '/' + uuid4(),
+            },
+        },
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _publish_repo(server_config, href, distributor_id, responses=None):
+    """Publish a repository.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param href: A string. The path to the repository to which a distributor
+        shall be added.
+    :param distributor_id: The ID of the distributor performing the publish.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return handle_response(requests.post(
+        server_config.base_url + href + 'actions/publish/',
+        json={'id': distributor_id},
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _gen_puppet_repo_body():
+    """Return a semi-random dict that used for creating an puppet repo."""
+    return {
+        'id': uuid4(),
+        'importer_config': {},
+        'importer_type_id': 'puppet_importer',
+        'notes': {'_repo-type': 'puppet-repo'},
+    }
+
+
+class _BaseTestCase(TestCase):
+    """Provide a server config, and tear down created resources."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Provide a server config and an iterable of resources to delete."""
+        cls.cfg = get_config()
+        cls.attrs_iter = tuple()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete created resources."""
+        for attrs in cls.attrs_iter:
+            delete(cls.cfg, attrs['_href'])
+
+
+class CreateTestCase(_BaseTestCase):
+    """Create two puppet repos, with and without feed URLs respectively."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create two puppet repositories, with and without feeds."""
+        super(CreateTestCase, cls).setUpClass()
+        cls.bodies = tuple((_gen_puppet_repo_body() for _ in range(2)))
+        cls.bodies[1]['importer_config'] = {'feed':
+                                            ''.join(['http://', uuid4()])}
```

Comment:
Is there any reason to use an `http://` prefix? This doesn't result in a valid base URL as is.